app/classifier/tasks/survey/editor.cjsx

```diff
@@ -247,24 +372,33 @@ module.exports = React.createClass
   makeID: (string) ->
     string.replace(/[aeiouy\W]/gi, '').toUpperCase()
 
-  ensureChoice: (name) ->
+  createChoice: (name) ->
     choiceID = @makeID name
-    unless choiceID in @props.task.choicesOrder
-      @props.task.choicesOrder.push choiceID
-    @props.task.choices[choiceID] ?=
+    unless choiceID in @state.task.choicesOrder
+      @state.task.choicesOrder.push choiceID
+    @state.task.choices[choiceID] ?=
       label: name
       description: ''
       noQuestions: false
       images: []
       characteristics: {}
       confusionsOrder: []
       confusions: {}
-    @props.task.choices[choiceID]
+    @state.task.choices[choiceID]
+
+  ensureChoice: (name) ->
+    choiceLabels = []
+    for choiceKey, choiceValue of @state.task.choices
+      choiceLabels.push choiceValue.label
+    noLabelMatch = choiceLabels.some (label, i) =>
+      label is name
+    unless noLabelMatch
+      throw new Error "#{name} does not have a Choices match"
 
   addChoice: ({name, description, no_questions, images, __parsedExtra}) ->
     unless name?
       throw new Error 'Choices require a "name" column.'
-    choice = @ensureChoice name
+    choice = @createChoice name
```

Comment:
with createChoice, ensureChoice, and addChoice, this code is really hard to read/understand